src/goto-instrument/contracts/dynamic-frames/dfcc_root_object.h

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************\
+
+Module: Dynamic frame condition checking
+
+Author: Remi Delmas, delmasrd@amazon.com
+
+Date: March 2023
+
+\*******************************************************************/
+
+/// \file
+/// Utility functions that compute root object expressions for assigns clause
+/// targets and LHS expressions.
+
+#ifndef CPROVER_GOTO_INSTRUMENT_CONTRACTS_DYNAMIC_FRAMES_DFCC_ROOT_OBJECT_H
+#define CPROVER_GOTO_INSTRUMENT_CONTRACTS_DYNAMIC_FRAMES_DFCC_ROOT_OBJECT_H
+
+#include <util/irep.h>
+
+#include <unordered_set>
+
+class exprt;
+class namespacet;
+class messaget;
+
+/// Computes a set of root object expressions for an assignment left hand side
+/// expression.
+std::unordered_set<exprt, irep_hash> dfcc_assignment_lhs_root_objects(
+  const exprt &lhs,
+  const namespacet &ns,
+  messaget &log);
+
+/// Computes a set of root object expressions for an assigns clause target
+/// expression that may contain disjunctions.
+std::unordered_set<exprt, irep_hash> dfcc_assigns_target_root_objects(
```

Comment:
Isn't it always safe to just use this one, so perhaps `dfcc_assignment_lhs_root_objects` should not be in the public interface?

Reply:
taking your suggestion